### ISSUE TITLE
Список ссылок пользователя

### ISSUE THREAD

**Issue by wladich** [wladich/nakarte#34]
Хранить надо на сервере, чтобы можно было переносить между устройствами
Надо найти сторонний сервис для авторизации, т.к. самому писать не хочется

**Comment by slazav** [comment]
С авторизацией через сторонние сервисы я игрался, пытался свой сайт переписывать разными способами. Вроде, рабочий код у меня есть и эта тема мне кажется не слишком страшной. 
А что значит "найти сторонный сервис"? Фейсбук, гугл, яндекс, вконтакте и т.д. по списку. Я еще через loginza делал. Разницы, на самом деле, не очень много.

**Comment by wladich** [comment]
Я с темой почти не знаком, даже не очень представляю, чем OAuth от OpenID отличается

**Comment by slazav** [comment]
Я это делал на очень простом уровне (видимо, это и называется oauth):
- Кнопка login ведет на специальный адрес в фейсбуке, гугле и т.п. В параметрах - некий мой url.
- Фейсбук (или другой сервис) спрашивает у пользователя что следует, и редиректит его на мой url с неким ключом.
- Дальше я на стороне сервера с помощью этого ключа (и еще специального секретного ключа)  спрашиваю у фейсбука информацию о результате авторизации и о пользователе.
- в случае удачи создаю сессию, храню ее в куках у пользователя и у себя в БД.

У меня это все тут: https://github.com/slazav/xyz_site/blob/master/cgi/login.pl
Я следовал этому примеру: https://ruseller.com/lessons.php?id=1674

**Comment by wladich** [comment]
А регистрировать свой сайт на гугле/фейсбуке надо?

**Comment by slazav** [comment]
Да, надо. Он, собственно и будет спрашивать "хотите ли вы выдать информацию сайту такому-то?"
На фейсбуке это, кстати, довольно многоступенчатый процесс, я долго пытался разобраться (и, кажется, до конца так и не довел)

**Comment by slazav** [comment]
С другой стороны, мне нравилась исходная идея nakarte.tk, что все происходит и хранится на стороне пользователя.

**Comment by wladich** [comment]
Ну переход на хранение треков на сервере уже сделан. В этом тикете речь именно об удобном поиске треков, на которые пользователь давал ссылку.

Хранение данных у пользователя мне тоже нравилось, но короткие ссылки победили. 
Если есть желание вернуть длинные ссылки в дополнение к коротким, то сделай новый тикет, там будем думать зачем оно надо и как сделать красиво.

**Comment by slazav** [comment]
Так чтобы найти "трек, на который пользователь давал ссылку" достаточно знать эту ссылку.

Переход я, конечно, заметил, и не знаю, хорошо ли это.  Но идея смешивать короткие и длинные ссылки еще более странная, мне кажется.

Я бы, пожалуй, не делал бы авторизацию. Мне бы хватило кнопок  "получить короткую ссылку", "сохранить в файл", "загрузить из файла", "загрузить из URL". Отдельно для каждого трека (тогда сохранять можно в GPX/KML, как сейчас) и отдельно для всех данных (тогда сохранять было бы удобно все состояние - в своем формате или в KML каком-нибудь, если туда можно все запихнуть, включая слои и т.п.).
+ возможность передавать url для загрузки через url.

**Comment by wladich** [comment]
Пользователи склонны терять эти самые ссылки. Я теряю.
Авторизация, естественно, по желанию -- позволит синхронизировать список ссылок между девайсами. А ещё можно и в localStorage хранить без авторизации.

Что-то у тебя сумбурно получилось.
Про состояние есть тикет https://github.com/wladich/nakarte/issues/33
Технически это тривиально, но я не понимаю, как это сделать в интерфейсе, чтобы и удобно и не загромождать имеющееся. Наверное, потому что мне это не особо нужно. Так что если заинтересован, попробуй про интерфейс туда написать.
И ещё есть про zip с треками https://github.com/wladich/nakarte/issues/32
